src/Db/Adapter/UnifiedMigrationsTableStorage.php

```diff
@@ -0,0 +1,270 @@
+            ]);
+
+        if ($this->plugin === null) {
+            $query->where([
+                'breakpoint !=' => 0,
+                'plugin IS' => null,
+            ]);
+        } else {
+            $query->where([
+                'breakpoint !=' => 0,
+                'plugin' => $this->plugin,
+            ]);
+        }
+
+        return $this->adapter->executeQuery($query);
+    }
+
+    /**
+     * Marks a migration as a breakpoint or not depending on $state.
+     *
+     * @param \Migrations\MigrationInterface $migration Migration
+     * @param bool $state The breakpoint state to set.
+     * @return void
+     */
+    public function markBreakpoint(MigrationInterface $migration, bool $state): void
+    {
+        $query = $this->adapter->getUpdateBuilder();
+        $query->update(self::TABLE_NAME)
+            ->set([
+                'breakpoint' => (int)$state,
+                'start_time' => $query->identifier('start_time'),
+            ]);
+
+        if ($this->plugin === null) {
+            $query->where([
+                'version' => $migration->getVersion(),
+                'plugin IS' => null,
+            ]);
+        } else {
+            $query->where([
+                'version' => $migration->getVersion(),
+                'plugin' => $this->plugin,
+            ]);
+        }
+
+        $this->adapter->executeQuery($query);
+    }
+
+    /**
+     * Creates the unified migration storage table.
+     *
+     * @return void
+     * @throws \InvalidArgumentException When there is a problem creating the table.
+     */
+    public function createTable(): void
+    {
+        try {
+            $options = [
+                'id' => true,
+                'primary_key' => 'id',
+            ];
+
+            $table = new Table(self::TABLE_NAME, $options, $this->adapter);
+            $table->addColumn('version', 'biginteger', ['null' => false])
+                ->addColumn('migration_name', 'string', ['limit' => 100, 'default' => null, 'null' => true])
+                ->addColumn('plugin', 'string', ['limit' => 100, 'default' => null, 'null' => true])
+                ->addColumn('start_time', 'timestamp', ['default' => null, 'null' => true])
+                ->addColumn('end_time', 'timestamp', ['default' => null, 'null' => true])
+                ->addColumn('breakpoint', 'boolean', ['default' => false, 'null' => false])
+                ->addIndex(['version', 'plugin'], ['unique' => true, 'name' => 'version_plugin_unique'])
+                ->save();
+        } catch (Exception $exception) {
+            throw new InvalidArgumentException(
+                'There was a problem creating the migrations table: ' . $exception->getMessage(),
+                (int)$exception->getCode(),
+                $exception,
+            );
+        }
+    }
+
+    /**
+     * Upgrades the migration storage table if needed.
+     *
+     * @return void
+     */
+    public function upgradeTable(): void
+    {
```

Comment:
Um why? We're creating a new table. Why wouldn't this be folded into the create method?